src/FillArrays.jl

```diff
@@ -265,4 +265,35 @@ Base.:*(a::ZerosVecOrMat, b::AbstractVector) = mult_zeros(a, b)
 Base.:*(a::AbstractVector, b::ZerosVecOrMat) = mult_zeros(a, b)
 Base.:*(a::ZerosVecOrMat, b::ZerosVecOrMat) = mult_zeros(a, b)
 
+
+# Unary = / -.
+Base.:+(a::Zeros) = a
+Base.:-(a::Zeros) = a
+
+# eltypes are the same, so result of promotion is guaranteed to be the same.
+function Base.:+(a::Zeros{T, N}, b::AbstractArray{T, N}) where {T, N}
```

Comment:
Is this method helpful? I would have thought that the `promote_type(T, V) == T ? a : ...` case below would compile down to the same thing.

Reply:
Good point, I hadn't thought about that. The following give identical results:
```julia
@code_warntype Zeros{Float64}(5, 4) + randn(5, 4)
@code_warntype Zeros{Float32}(5, 4) + randn(5, 4)
```
which suggests that you are correct, but I'm not entirely confident I've not missed something subtle. If you think this confirms what you're saying, I'll make the appropriate changes.

Reply:
I'm not sure either, but it seems best to start with fewer methods, then add the extra method if an exception is found.